Source/engine/point.hpp

```diff
@@ -58,13 +59,27 @@ struct Point {
 		return (*this) += Point::fromDirection(direction);
 	}
 
+	constexpr Point &operator+=(const Size &size)
+	{
+		x += size.width;
+		y += size.height;
+		return *this;
+	}

```

Comment:
I'm still not a super fan of this, but as long as Size isn't negative I suppose it's ok. Displacement is much preferred and I don't like the idea of starting to introduce Size in places that we already know we want to change in the near future.